Server/src/services/track.service.ts

```diff
@@ -209,7 +222,11 @@ export default class TrackService {
 			// this gives us the nearest point on the linestring including the distance to that point
 			const closestPoint: GeoJSON.Feature<GeoJSON.Point> = nearestPointOnLine(lineStringData, position)
 			if (closestPoint.properties == null || closestPoint.properties["dist"] == null) {
-				// TODO: this should not happen, so maybe log this
+				logger.warn(
+					`Turf did not calculate nearest point on line correctly for position ${JSON.stringify(
```

Comment:
Same as above: 
Turf error: Could not calculate nearest point on line correctly for position ${JSON.stringify(position)} for track with id {tracks[i].uid}.

I think it is just easier to read that this error lies in turf, not our implementation.

Reply:
I was not thinking too much about this as this check should prevent typescript from complaining (the `properties`-field is optional in GeoJSON) rather than really handling any errors there. I guess turf would throw an error instead of setting `properties` to `null` anyways. But you are right, that the log messages are a little bit vague. So I will change this (thanks for the suggestions for those).